textalyze.js

```diff
@@ -29,40 +31,116 @@ function itemCounts(array) {
  * @returns {Array} The Array containing the characters from the string
  */
 function arrayFrom(string) {
-  if (validateString(string)) {
+  if (!isString(string)) {
     return [];
   }
 
   return string.split('');
 }
 
 /**
- * Given an input String, returns the string with only lower case characters
+ * Given an input String, returns the string with only lower case a-z characters
  * @param {String} string - The String to be sanitized
  * @returns {String} The String sanitized
  */
 function sanitize(string) {
-  if (validateString(string)) {
+  if (!isString(string)) {
     return '';
   }
 
-  return string.toLowerCase();
+  return string.toLowerCase().replace(/[^a-z]/gi, '');
+}
+
+/**
+ * Given an input Map and total count of elements,
+ * returns the object containing the percentage acording to total
+ * @param {Map} counts - The map containing the counts for each value
+ * @param {Number} total - The total count of elements
+ * @param {Boolean} sorted - If the return must be sorted
+ * @returns {Map} The frequency in percentage of each value
+ */
+function getFrequencyStatistics(counts, total, sorted) {
+  if (!counts || !(counts instanceof Map)) {
+    return counts;
+  }
+
+  if (!total || Number.isNaN(Number(total))) {
+    return counts;
+  }
+
+  let statistics = new Map();
+
+  counts.forEach((value, key) => {
+    const percentage = value / total;
+    statistics.set(key, percentage);
+  });
+
+  if (sorted) {
+    statistics = new Map([...statistics.entries()].sort());
+  }
+
+  return statistics;
 }
 
-module.exports = { itemCounts, arrayFrom, sanitize };
+/**
+ * Builds a string containing repetitions of histogramString
+ * according to current value in comparison to maxValue
+ * @param {Number} value - The current value
+ * @param {Number} maxValue - The max value for param value
+ * @param {String} histogramString - The string to compose the histogram bar
+ * @param {Number} maxBarLength - The max amount of times histogramString can be repeated
+ * @returns {String} The hitogram bar
+ */
+function getHistogramBar(value, maxValue, histogramString, maxBarLength) {
+  const charCount = value * maxBarLength / maxValue;
+  const histogramBar = histogramString.repeat(charCount);
+
+  return histogramBar;
+}
+
+module.exports = {
+  itemCounts,
+  arrayFrom,
+  sanitize,
+  getFrequencyStatistics,
+  getHistogramBar,
+};
 
 //
 // running the app
 //
-if (require.main === module) {
-  const string = 'Hello World';
+/**
+ * Runs the main application
+ * @param {String []} args - The args
+ */
+function main(argv) {
```

Comment:
I actually liked how you had it before, with stripping off the first two arguments.  [Node gurantees])https://nodejs.org/docs/latest/api/process.html#process_process_argv) that the first element of `process.argv` is always `process.execPath` and the second element is the name of the script being executed.

So, unless `main` needs those values for some reason, I think it's fine to toss them out.  This would also make testing less tedious if you decide to test `main`.  Imagine having to always do this:

```javascript
const args = [null, null, 'dummy_file.txt', '='];
expect(main(args))...
```

Versus

```javascript
const args = ['dummy_file.txt', '='];
expect(main(args))...
```

Removing the first two elements also means you could do

```javascript
function main(path, histogramString = '-') {
  ...
}
```

which would mean someone wouldn't have to read the actual code of `main` to see what the expectations are for the arguments.